src/utils/products.utils.ts

```diff
@@ -14,7 +18,81 @@ export const getFilteredProducts = (
   });
 };
 
-export const getStarsArrFromNumber = (num: number) => {
-  const arr = Array(5).fill(false);
+export const getStarsArrFromNumber = (num: number): boolean[] => {
+  const arr: boolean[] = Array(5).fill(false);
   return arr.map((_, index) => index < num);
 };
+
+export const getCategories = (productsArr: IProduct[]): string[] => {
+  return productsArr.reduce((acc: string[], product) => {
+    if (!acc.includes(product.category)) {
+      acc.push(product.category);
+    }
+    return acc;
+  }, []);
+};
+
+export const getCategoriesWithCount = (
+  productsArr: IProduct[]
+): ICategoryWithCount[] => {
+  const categories: ICategoryWithCount[] = productsArr.reduce(
+    (acc: ICategoryWithCount[], product) => {
+      const existingCategory = acc.find(
+        (category) => category.title === product.category
+      );
+      if (existingCategory) {
+        existingCategory.count++;
+      } else {
+        acc.push({ title: product.category, count: 1 });
+      }
+      return acc;
+    },
+    []
+  );
+  return categories;
+};
```

Comment:
Can't you use categories list from 'getCategoriesWithCount' ?

Reply:
Fixed - create only 1 function that returns { [category]: { count: number, brands: string[] } }. And from that obj i get categories, brands and count